bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -390,20 +392,37 @@ public void unshift(Object[] values) {
 
     @Override
     public String stringValue(BLink parent) {
-        StringJoiner sj = new StringJoiner(" ");
+        StringJoiner sj = new StringJoiner(",");
         for (int i = 0; i < this.size; i++) {
-            sj.add(StringUtils.getStringValue(this.refValues[i], new CycleUtils.Node(this, parent)));
+            Object value = this.refValues[i];
```

Comment:
for a `nil` value `toString()` should be `null`. It should be specifically handled here. 
```ballerina
[int?, string] a = [(), "ABC"];
```
`a.toString()` is `[,"ABC"]` ATM, but should be `[null,"ABC"]`